src/otx/v2/adapters/torch/lightning/visual_prompt/modules/datasets/dataset.py

```diff
@@ -0,0 +1,392 @@
+            shuffle=False,
+            batch_size=self.config.val_batch_size,
+            num_workers=self.config.num_workers,
+            collate_fn=collate_fn,
+        )
+
+    def test_dataloader(self) -> DataLoader | list[DataLoader]:
+        """Test Dataloader.
+
+        Returns:
+            Union[DataLoader, list[DataLoader]]: Test Dataloader.
+        """
+        return DataLoader(
+            self.test_dataset,
+            shuffle=False,
+            batch_size=self.config.test_batch_size,
+            num_workers=self.config.num_workers,
+            collate_fn=collate_fn,
+        )
+
+    def predict_dataloader(self) -> DataLoader | list[DataLoader]:
+        """Predict Dataloader.
+
+        Returns:
+            Union[DataLoader, list[DataLoader]]: Predict Dataloader.
+        """
+        return DataLoader(
+            self.predict_dataset,
+            shuffle=False,
+            batch_size=1,
+            num_workers=self.config.num_workers,
+            collate_fn=collate_fn,
+        )
+
+
+class VisualPromptInferenceDataset(Dataset):
+    """Inference Dataset to perform prediction.
+
+    Args:
+        path (str | Path): Path to an image or image-folder.
+        transform (A.Compose | None, optional): Albumentations Compose object describing the transforms that are
+            applied to the inputs.
+        image_size (int | tuple[int, int] | None, optional): Target image size
+            to resize the original image. Defaults to None.
+    """
+
+    def __init__(
+        self,
+        path: str | Path,
+        transform: al.Compose | None = None,
+        image_size: int = 1024,
+    ) -> None:
+        """Initializes a Dataset object.
+
+        Args:
+            path (str or Path): The path to the directory containing the images.
+            transform (al.Compose or None): A composition of image transformations to apply to the images.
+                Defaults to None.
+            image_size (int): The size of the images to be loaded. Defaults to 1024.
+        """
+        super().__init__()
+
+        self.image_filenames = get_image_filenames(path)
+
+        if transform is None:
+            self.transform = get_transform(image_size=image_size)
+        else:
+            self.transform = transform
+
+    def __len__(self) -> int:
+        """Get the number of images in the given path."""
+        return len(self.image_filenames)
+
+    def __getitem__(self, index: int) -> dict:
+        """Get dataset item.
+
+        Args:
+            index (int): Index of the dataset sample.
+
+        Returns:
+            dict: Dataset item.
+        """
+        image_filename = self.image_filenames[index]
+        image = read_image(path=image_filename)
+        item: dict = {"index": index, "images": image}
+
+        prompts = {
+            "original_size": image.shape,
+            "gt_masks": [],
+            "bboxes": [],
+            "points": [],
+            "labels": [],
```

Comment:
Current visual prompting task only enables inference with prompts, such as bboxes and points, so they must be included in `prompts` dict.

Reply:
What do you think I should be doing here? Or should I just leave it as is for now and check it out later?

Reply:
When I think it again, leaving it as is would be better.
Fine-tuning uses an image and those prompts together, so that dataset for inference is unnecessary.
But zero-shot learning to be implemented later can use that dataset with a few modifications because zsl uses only an image.

Reply:
Okay, I'm going to leave this as is. thanks!